drivers/adc/adc_ad5592.c

```diff
@@ -222,9 +244,11 @@ static int adc_ad5592_init(const struct device *dev)
 			(k_thread_entry_t)adc_ad5592_acquisition_thread, data, NULL, NULL,
 			CONFIG_ADC_AD5592_ACQUISITION_THREAD_PRIO, 0, K_NO_WAIT);
 
-	ret = k_thread_name_set(tid, "adc_ad5592");
-	if (ret < 0) {
-		return ret;
+	if (IS_ENABLED(CONFIG_THREAD_NAME)) {
+		ret = k_thread_name_set(tid, "adc_ad5592");
+		if (ret < 0) {
+			return ret;
+		}
```

Comment:
Wouldn't it be better to ignore the return value of `k_thread_name_set`? That way it will work fine when thread naming is off but tracing is on; and when both are on or off.

Reply:
Result can be ignored and it is done this way in number of files. After analysis of the `k_thread_name_set()` function one can say that the return value is useles (and it is in fact at the moment). On the other hand this is how the API looks like and my approach here was to be aligned to the API well. If dropping this function's result is preffered in Zephyr I will change the code. Nevertheless I would vote for changing the API then :) 

Regarding the tracing, if `CONFIG_TRACING` is disabled the `SYS_PORT_TRACING_OBJ_FUNC` macro is defined as:

```
#define SYS_PORT_TRACING_OBJ_FUNC(obj_type, func, obj, ...) do { } while (false)
```

So it should not cause issues. Am I missing some puzzle here?

Reply:
> Wouldn't it be better to ignore the return value of `k_thread_name_set`? That way it will work fine when thread naming is off but tracing is on; and when both are on or off.

For some reason, the `k_thread_name_set` function returns a specific return code we should handle. The suggested solution with `IS_ENABLED` macro is pretty enough for me.

Reply:
> Nevertheless I would vote for changing the API then :)

+1

Reply:
I've looked at the other PRs and Zephyr sources and it seems that the more common approach is to simply ignore the return code from `k_thread_name_set` even if thread names are enabled. This is a bit of nonsense and I agree that changes need to be made to the API.